File: 2519-find-the-original-array-of-prefix-xor/2519-find-the-original-array-of-prefix-xor.py
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        start=0
        end=len(pref)-1
        res=[pref[0]]
        while(start<end):
            res.append(pref[start]^pref[start+1])
            start+=1
        return res


        

File: 2519-find-the-original-array-of-prefix-xor/2519-find-the-original-array-of-prefix-xor.cpp
class Solution {
public:
    vector<int> findArray(vector<int>& pref) {
        int p=0,i=0,n=pref.size();
        vector<int> res;
        while(i<n)
        {
            res.push_back(pref[i]^p);
            p=pref[i];
            i++;
        }
        return res;
        
    }
};